src/Components/Common/ErrorCatcher/ErrorCatcher.tsx

```diff
@@ -1,23 +1,29 @@
-import type { Dispatch, ReactElement, ReactNode, SetStateAction } from 'react';
+import type { ReactElement, ReactNode } from 'react';
 import { Component } from 'react';
 import { Error } from '../Error/Error';
 import { ErrorPopup } from '../ErrorPopup/ErrorPopup';
 import type { ErrorType, Nullable } from '../../../utils/types/common';
 
 type PropsType = {
   children: ReactNode;
-  globalError: Nullable<ErrorType>;
-  setGlobalError: Dispatch<SetStateAction<Nullable<ErrorType>>>;
 };
 type StateType = {
-  errorMessage: string;
+  UIError: Nullable<ErrorType>;
+  promiseError: Nullable<ErrorType>;
 };
 
 export class ErrorCatcher extends Component<PropsType, StateType> {
-  state = { errorMessage: '' };
+  state: StateType = {
+    UIError: null,
+    promiseError: null,
+  };
 
   static getDerivedStateFromError() {
-    return { errorMessage: 'Some UI Error! We are sorry... Fix it soon!' };
+    return {
+      UIError: {
+        message: 'Some UI Error! We are sorry... Fix it soon!',
```

Comment:
Be more polite here =)
We are sorry... We will fix it soon! or 
We are sorry... The fix is coming!

Reply:
Done!
I  am just learning English... :)